drivers/sx127x/sx127x_getset.c

```diff
@@ -113,11 +113,34 @@ void sx127x_set_syncword(sx127x_t *dev, uint8_t syncword)
     sx127x_reg_write(dev, SX127X_REG_LR_SYNCWORD, syncword);
 }
 
+/**
+ * The carrier frequency is given by Fc = Fstep * RFR (HW value).
+ * Since Fstep is (32 MHz) / (2^19), these functions below use
+ * bit shifting and integer division to convert between reg
+ * value and the actual carrier frequency.
+ */
+
+/**
+ * Convert from carrier frequency (Hz) to HW value.
+ */
+static inline uint32_t _freq_to_hw_value(uint32_t freq)
+{
+    return (freq << 8) / 15625;
```

Comment:
```suggestion
    return (freq / 15625) << 8;
```
I had no idea `freq` would be this big.

Reply:
Hmmmm this can produce an offset of 256 in the Hw value :(

Given `a=int(freq / 15625)` the integer division and `b=(freq mod 15625)/15625` the decimal reminder (so `freq/15625 == a+b`, with 0<=b<1)

Then (1):
`(freq/15625) * 2^8 == (a + b) * 2^8 == a*2^8 + b* 2^8`

But (2):
`int(freq/15625)*256 == a*256.`

So `(1) - (2)` is `b*256`.

I tested it with real numbers:
```
>>> freq
16777215
>>> int(freq<<8)/(5**6)
274877.89056
>>> int(freq/(5**6))<<8
274688
```

Reply:
Hmm, how about

```C
static inline uint32_t _freq_to_hw_value(uint32_t freq)
{
    freq <<= 2;

    unsigned rounds = 6;
    do {
        freq /=5;
        freq <<= 1;
    } while (--rounds);

    return freq;
}
```

Reply:
This gets much better, but generates a difference of "2" in worst scenario (120 Hz in the carrier frequency).
I have the impression it shouldn't affect that much (it's 0.00001% of drift in the 868 MHz band).